collect_app/src/main/java/org/odk/collect/android/utilities/DateTimeUtils.java

```diff
@@ -107,12 +107,12 @@ private static String getCustomDateTimeLabel(Date date, DatePickerDetails datePi
     private static String getDateTimePattern(boolean containsTime, DatePickerDetails datePickerDetails) {
         String datePattern;
         if (containsTime) {
-            datePattern = "yyyyMMMdd HHmm";
+            datePattern = "yyyyMMMMdd HHmm";
```

Comment:
I'm just trying to understand what is happening here.  Can you explain this change?  It's used by `getGregorianDateTimeLabel` for Android versions equal to Jelly Bean or higher.

It looks like it would produce a Gregorian label like `2018November15 12:34` (or below, `2018November15` or `2018November`).  Why do we want a label with the month name squished next to the year like that?  Or have I misunderstood what this will do?

Reply:
`getDateTimePattern()` is used to get a datePattern (Skeleton) that is passed to `android.text.format.DateFormat.getBestDateTimePattern()` _refer line 45_

The class then returns as per https://developer.android.com/reference/android/text/format/DateFormat#getBestDateTimePattern(java.util.Locale,%20java.lang.String)

Quoting the Docs

> public static String getBestDateTimePattern (Locale locale, 
                String skeleton)

> Returns the best possible localized form of the given skeleton for the given locale. A skeleton is similar to, and uses the same format characters as, a Unicode UTS #35 pattern.
> 
> One difference is that order is irrelevant. For example, "MMMMd" will return "MMMM d" in the en_US locale, but "d. MMMM" in the de_CH locale.
> 
> Note also in that second example that the necessary punctuation for German was added. For the same input in es_ES, we'd have even more extra text: "d 'de' MMMM".
> 
> This method will automatically correct for grammatical necessity. Given the same "MMMMd" input, this method will return "d LLLL" in the fa_IR locale, where stand-alone months are necessary. Lengths are preserved where meaningful, so "Md" would give a different result to "MMMd", say, except in a locale such as ja_JP where there is only one length of month.
> 
> This method will only return patterns that are in CLDR, and is useful whenever you know what elements you want in your format string but don't want to make your code specific to any one locale.

Reply:
Oh, I understand now.  Thank you for explaining.  The variable name `datePattern` is misleading, then—this string is not a date pattern; it is a skeleton.  Could you rename `datePattern` to `skeleton`?